src/main/scripts/src/app/shared/perspectives/documents/table/table-documents-perspective.component.ts

```diff
@@ -18,23 +18,135 @@
 })
-export class TableDocumentsPerspectiveComponent implements Perspective {
+export class TableDocumentsPerspectiveComponent implements Perspective, OnInit {
 
   @Input()
   public query: string;
 
   @Input()
   public editable: boolean;
 
-  constructor(private documentService: DocumentService) {
+  public collection: Collection;
+  public header: TableHeader;
+  public rows: TableRow[];
+
+  public settings = <TableSettings>{
+    color: '#3498DB',
+    highlightColor: '#F39C12',
+    editable: true,
+    lineNumberColor: '#b4bcc2'
+  };
+
+  constructor(private collectionService: CollectionService,
+              private documentService: DocumentService,
+              private route: ActivatedRoute) {
+  }
+
+  public ngOnInit(): void {
+    this.route.paramMap
+      .switchMap(params => {
+        let code: string = params.get('collectionCode');
+
+        return Observable.combineLatest([
+          this.collectionService.getCollection(code),
+          this.collectionService.getAttributes(code),
+          this.documentService.getDocuments(code)
+        ]);
+      }).subscribe(([collection, attributes, documents]) => {
+      this.collection = collection;
+      this.prepareTableData(attributes, documents);
+    });
+  }
+
+  public onNewValue(data: any) {
+    let copy = Object.assign({}, data);
+    let rowIndex: number = copy.rowIndex;
+    delete data.rowIndex;
+    this.documentService.createDocument(this.collection.code, this.convertTableDataToDocument(data))
+      .subscribe((json: object) => this.rows[rowIndex].id = json['_id']);
+  }
+
+  public onValueChange(data: any) {
+    this.documentService.updateDocument(this.collection.code, this.convertTableDataToDocument(data))
+      .subscribe();
+  }
+
+  public onHeaderChange(data: any) {
+    let oldValue: string = data.oldValue;
+    let newValue: string = data.newValue;
+    this.collectionService.renameAttribute(this.collection.code, oldValue, newValue)
+      .subscribe();
+  }
+
+  public onRemoveColumn(columnName: string) {
+    this.collectionService.dropAttribute(this.collection.code, columnName)
+      .subscribe();
+  }
+
+  public onDragColumn(data: any) {
+    // TODO
+  }
+
+  public onHideColumn(data: any) {
+    // TODO
+  }
+
+  public onShowColumn(data: any) {
+    // TODO
+  }
+
+  private convertTableDataToDocument(data: any): Document {
+    let document: Document = new Document();
+    Object.keys(data).forEach(key => {
+      if (key === 'id') {
+        document.id = data[key];
+      } else {
+        document.data[key] = data[key];
+      }
+    });
+    return document;
+  }
+
+  private prepareTableData(attributes: Attribute[], documents: Document[]) {
+    let headerRows: TableHeaderCell[] = attributes.map(attr =>
+      <TableHeaderCell>{label: attr.name, active: false, hidden: false, constraints: attr.constraints});
+    this.header = <TableHeader> {cells: headerRows};
+    this.rows = documents.map(document => {
```

Comment:
This method is too complicated and should be split into multiple simpler methods.